docs/pages/x/api/date-pickers/localization-provider.json

```diff
@@ -1,6 +1,6 @@
 {
   "props": {
-    "adapterLocale": { "type": { "name": "union", "description": "object<br>&#124;&nbsp;string" } },
```

Comment:
I think having `any` here is good, because before we were kind of lying.
We were saying "string | object" when in fact you could only use one of the two (depending on your adapter)
With "any" at least it's clear that we did not type this prop on the proptypes